syntax = "proto3";

package permissions;

service Permissions {
    rpc CheckEmailVerifiedPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
    rpc CheckCustomerPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
    rpc CheckCourierPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
    rpc CheckRestaurantManagerPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
    rpc CheckModeratorPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
}

message CheckPermissionRequest {
    string access_token = 1;
}

message CheckPermissionResponse {
    bool has_permission = 1;
    int64 user_id = 2;
}
